### PR TITLE
use gatsby to generate a static website.

### LINKED ISSUE #4: Generate static website
## Context

Our static website is buit on gh-pages with this npm task `"deploy": "gh-pages -d build",`. This is a not a true static website and it's bad for SEO.

We need something to generate this :). @lutangar will be the reviewer

### PR DESCRIPTION
**Pull request by JulesAaelio**
use gatsby to generate a static website.

## Description
Use gatsby to generate a static website, update docker
configuration to use gatsby live development server and
update README.md with installation/usage commands. Update
package.json so the website is build and sent to gh-pages.

## Related Issue
close #4 

## Motivation and Context
See #4

## How Has This Been Tested?
The result has been compared to actual online version of the website by a human eye. 
Deployment has not been tested yet. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Test deployment to GitHub pages.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +20,7 @@
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
+
+# Build directory
+/public
+.DS_Store
```
Comment (pocky) [inline .gitignore]: still nop :D (global .gitignore)

File: package.json [hunk 2]
```diff
@@ -1,25 +1,32 @@
 {
```
Comment (pocky) [inline package.json]: Too much copy/paste in this file :D

File: src/components/Header.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React from "react";
+import React from 'react'
```
Comment (pocky) [inline src/components/Header.js]: nop or single quote everywhere

File: .gitignore (hunk 1 above)
Comment (lutangar) [inline .gitignore]: .DS_Store is already present in the file

File: src/components/Header.js [hunk 4]
```diff
@@ -10,11 +10,11 @@ class Header extends React.Component {
   render() {
     return (
       <header>
-        <h1 itemprop="name">{this.props.title}</h1>
-        <em itemprop="description">{this.props.baseline}</em>
+        <h1 itemProp="name">{this.props.title}</h1>
```
Comment (pocky) [inline src/components/Header.js]: itemprop != itemProp (same for itemscope)

File: .prettierrc [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "semi": false,
+  "singleQuote": true,
+  "trailingComma": "es5"
```
Comment (lutangar) [inline .prettierrc]: I'd like we add a `print-width` of 120 here.

File: src/components/Header.js [hunk 6]
```diff
@@ -10,11 +10,11 @@ class Header extends React.Component {
...
+        <em itemProp="description">{this.props.baseline}</em>
       </header>
     );
   }
 }
 
-export default Header;
+export default Header
```
Comment (pocky) [inline src/components/Header.js]: Missing ";" or no ";" at all.

File: src/components/Social.js [hunk 7]
```diff
@@ -6,6 +6,7 @@ class Social extends React.Component {
     title: PropTypes.string.isRequired,
     icon: PropTypes.string.isRequired,
     href: PropTypes.string.isRequired
+
```
Comment (pocky) [inline src/components/Social.js]: remove this line

File: Makefile [hunk 8]
```diff
@@ -37,7 +37,7 @@ yarn-build: ## Build and deploy project on your host
 deploy:
 	yarn deploy
 
-.PHONY: kill install reset start stop clean deploy no-docker
+.PHONY: kill install reset start stop clean deploy no-docker build
```
Comment (lutangar) [inline Makefile]: One command is missing from the list.

File: src/components/layout.js [hunk 9]
```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Helmet from 'react-helmet'
+import Footer from './Footer'
+
+const Layout = ({ children }) => (
+      <>
```
Comment (pocky) [inline src/components/layout.js]: 4 spaces or 2 spaces?

File: gatsby-config.js [hunk 10]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+  siteMetadata: {
+    title: 'Gatsby Default Starter',
```
Comment (lutangar) [inline gatsby-config.js]: We must change the metadata with the ones used in the previous site version.

File: package.json [hunk 11]
```diff
@@ -1,25 +1,32 @@
...
-  "name": "lepetitbloc.github.io",
-  "version": "0.1.0",
-  "private": true,
-  "homepage": "https://lepetitbloc.net",
+  "name": "gatsby-starter-default",
```
Comment (lutangar) [inline package.json]: all of these should be changed according to the project

File: public/index.html [hunk 12]
```diff
@@ -1,27 +1 @@
...
-
-<body itemscope itemtype="https://schema.org/WebPage">
-<noscript>
-    You need to enable JavaScript to run this app.
-</noscript>
-<div id="root" role="main"></div>
-</body>
-
-</html>
+<!DOCTYPE html><html><head><meta charSet="utf-8"/><meta http-equiv="x-ua-compatible" content="ie=edge"/><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"/><meta name="theme-color" content="#000000"/><link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.0/css/bootstrap.min.css"/><link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:600,700,900"/><title data-react-helmet="true"></title><script src="/socket.io/socket.io.js"></script></head><body><div id="___gatsby"></div><script src="/commons.js"></script></body></html>
```
Comment (lutangar) [inline public/index.html]: This file should not be committed.

File: src/components/Footer.js [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+    <div className="container-fluid text-center">
+      <p>
+        Crafted with{" "}
+        <span role="img" aria-label="love">
+                ❤️
+              </span>, <a href="https://reactjs.org/">react</a>,{" "}
+        <a href="http://getbootstrap.com/">bootstrap</a> and{" "}
+        <a href="https://fontawesome.com/">font-awesome</a> in{" "}
+        <span role="img" aria-label="france">
+                🇫🇷
```
Comment (lutangar) [inline src/components/Footer.js]: You must run prettier before committing/pushing anything:
`npm run format`

File: src/components/Home.js [hunk 14]
```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import Social from './Social'
+import People from './People'
+import TimeCard from './TimeCard'
+import content from '../content'
+import Header from './Header'
+import './Home.modules.css'
+
+const Home = () => (
+  <div>
```
Comment (lutangar) [inline src/components/Home.js]: If the `<div` tag doesn't bring any value, you should use a react fragment instead.

File: src/components/Social.js (hunk 7 above)
Comment (lutangar) [inline src/components/Social.js]: this extra line is not required

File: src/components/layout.js [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+        <Helmet
+          title={"LePetitBloc - Build a decentralised, trustable and reliable world"}
```
Comment (lutangar) [inline src/components/layout.js]: If possible, we should use the gatsby way for metas instead.

**Comment by lutangar** [comment]
@JulesAaelio @pocky all coding style issues will be solved with the right prettier config

File: src/components/Header.js (hunk 4 above)
Comment (JulesAaelio) [reply src/components/Header.js]: According to https://reactjs.org/docs/dom-elements.html, 

```
In React, all DOM properties and attributes (including event handlers) should be camelCased. For example, the HTML attribute tabindex corresponds to the attribute tabIndex in React. The exception is aria-* and data-* attributes, which should be lowercased. For example, you can keep aria-label as aria-label.
```

The attributes are lowercased on rendering.

File: gatsby-config.js [hunk 16]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+  siteMetadata: {
+    title: `LePetitBloc - Build a decentralised, trustable and reliable world`,
```
Comment (lutangar) [inline gatsby-config.js]: Use simple quotes instead of backticks if you don't need string interpolation.

File: src/components/Home.js [hunk 17]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import Social from './Social';
+import People from './People';
+import TimeCard from './TimeCard';
+import content from '../content';
+import Header from './Header';
+import './Home.modules.css';
+
+const Home = () => (
+  <React.Fragment>
```
Comment (lutangar) [inline src/components/Home.js]: There is a simpler synthax for React fragments: 
```
<>
// ...
</>
```
> /!\ It may not works for older React versions

**Pull request merged**